In [1]:
import pandas as pd
import numpy as np

### 1.A number is a Disarium number if the sum of the digits raised to the power of its respective position in the original number is equal to that number itself. Which of the following is the Disarium number ?
89 175 123` 131


In [2]:
#consider a number from user
number=input("Enter any number: ")

#save the given number in a list
x=list(number)

Enter any number: 131


In [4]:
# traverse the list , apply exponents according to the positions for each digit in a number and find its sum
k=0
for i in range(0,len(x)):
      y=int(x[i])**(i+1)
      k=k+y

In [5]:
#compare the obtained result with the given number 
if(k==int(number)):
    print(number,"is a Disarium number") 
else:
    print(number,"is  not a Disarium number")

131 is  not a Disarium number


### 1.Ans) Hence 89,175 are disarium numbers. 123,131 are not disarium numbers

In [16]:
#...............................................................................

### 2. Right now the time is 3:00 PM, if you will add 70 more hours, what would be the time then?

In [11]:
#function to convert 12hrs time format to 24hrs time format:

def convert_to_24hrs_format(time):
    if time[:2]=='12' and time[-2:]=='AM':
        return '00'+time[2:-2]
    elif time[-2:]=='AM':
        return time[:-2]
    elif time[-2:]=='PM':
        return str(int(time[:2])+12)+time[2:-2]
    else:
        return 'entered in wrong format'
    

In [12]:
#function to convert 24hrs time format to 2hrs time format:

def convert_to_12hrs_format(time):
    if int(time[:2])>12 :
        return str(int(time[:2])-12)+time[2:]+' PM'
    elif int(time[:2])==00:
        return '12'+time[2:]+' AM'
    else:
        return time[::]+' AM'
    

In [15]:
# consider time from user
print("Enter time in 12 hrs time format [HH:MM:SS AM/PM] ")
Given_time=input()

#consider the hours to be incremented
print("Enter number of hours to be incremented :")
change=int(input())

#convert the given time to 24hrs time  format 
x=convert_to_24hrs_format(Given_time)

#increment and apply mod . Save the obtained time in 24hrs time format in a variable
y=int(x[:2])+change
y=y%24
final_time=str(y)+x[2:]


#converting it to 12 hrs time format
final_time=convert_to_12hrs_format(final_time)
print("Required time :",final_time)

Enter time in 12 hrs time format [HH:MM:SS AM/PM] 
03:00:00 PM
Enter number of hours to be incremented :
70
Required time : 1:00:00  PM


## 2.Ans  if  70 more hours are added to 3:00:00 PM then the time is 1:00:00  PM

#..........................................................................................

 ### 3. A cricket player's statistics in the last 10 games are :__
 
     Runs Scored : 23 , 56, 67, 9, 56*, 90, 41, 40, 104, 3
     
     Balls Played : 45, 46, 90, 4, 44, 71, 70, 34, 81, 7

### What is the average number of runs scored by the player in the last 10 games ?

###  What is the strike rate of a player in the last 10 games?


In [19]:
#consider an empty list
Runs_scored=[]

#append user values(10) to the 'Runs_scored' list
print("Enter the number of runs scored by the player in the last 10 games")
for i in range(10):
    x=int(input())
    Runs_scored.append(x)
    
#print the list for confirmation
print ("Runs_scored: ",Runs_scored)

Enter the number of runs scored by the player in last 10 games
23
56
67
9
56
90
41
40
104
3
Runs_scored:  [23, 56, 67, 9, 56, 90, 41, 40, 104, 3]


In [24]:
#consider an empty list
Balls_Played=[]

#append user values(10) to the 'Balls_Played' list
print("Enter the number of balls played by the player in the last 10 games")
for i in range(10):
    x=int(input())
    Balls_Played.append(x)
    
print ("Balls_Played: ",Balls_Played)

Enter the number of balls played by the player in the last 10 games
45
46
90
4
44
71
70
34
81
7
Balls_Played:  [45, 46, 90, 4, 44, 71, 70, 34, 81, 7]


In [28]:
#find average of runs_scored by adding all the scores in the list(Runs_scored) and dividing by total number of elements(10)
Average=sum(Runs_scored)/10
print("Average number of runs scored by the player in the last 10 games:  ",Average)

Average number of runs scored by the player in the last 10 games:   48.9


In [36]:
# Strike Rate = (Runs_scored/ Balls_Played) * 100
Strike_Rate=(sum(Runs_scored)/sum(Balls_Played))*100
print("The strike rate of the required player in the last 10 games:",Strike_Rate)

The strike rate of the required player in the last 10 games: 99.39024390243902


### 3.Ans, a.Average number of runs scored by the player in the last 10 games:   48.9
###       b.The strike rate of the required player in the last 10 games: 99.39

    

In [ ]:
#...............................................................................................

### 4. What is the value of the expression 100 / 25

4, 4.0, Both, None


In [4]:
x=100 / 25
print(x)
type(x)

4.0


float

### 4.Ans, The value of the expression 100 / 25 is 4.0

In [ ]:
#........................................................................................................

## 5. What is the output of (1.1 + 2.2 == 3.3) ?
True,   False



In [5]:
1.1+2.2

3.3000000000000003

In [6]:
1.1 + 2.2 == 3.3

False

### 5.Ans, FALSE

In [ ]:
#...................................................................................................

### 6. Create a variable x with the value 10. Increase the value of x tenfold using an augmented assignment operator?

In [15]:
x=10
x*=10
print(x)

100


### 6.Ans,100

In [18]:
#...............................................................................

###  7.  Which of the following are valid Python variable names ?


Var1.3,

4tower,

Age,

Upi_id,

Upi66,

Return


In [24]:
Var1.3= 3  #Error as special character period exists

SyntaxError: invalid syntax (<ipython-input-24-6927dcdec2c5>, line 1)

In [26]:
4tower= 4 #Error as variable name starting with a digit

SyntaxError: invalid syntax (<ipython-input-26-786f581564be>, line 1)

In [28]:
Age = 23

In [29]:
Upi_id=56

In [30]:
Upi66 = 76

In [32]:
Return=90 # it is valid variable name as Return is not a keyword (where return is a keyword)

###  7,Ans:      Age, Upi_id, Upi66, Return are valid Python variable names

In [38]:
#.........................................................................................

### 8.What is the value of Temperature 101.5 degree Celsius into degree Fahrenheit 

In [41]:
#consider the temerature in degree celsius from user
temperature_celsius=float(input("Enter temperature in degree Celsius: "))

Enter temperature in degree Celsius: 101.5


In [45]:
#apply the conversion formula 
temperature_fahrenheit=(temperature_celsius*1.8)+32
print("Temperature in degree Fahrenheit: ",temperature_fahrenheit)

Temperature in degree Fahrenheit:  214.70000000000002


### 8, Ans: Hence Temperature in degree Fahrenheit:  214.7

In [ ]:
#............................................................................

### 9.What will be the value for x after executing the following code :-

x = 0

a = 0

b = -5

if a > 0:

    if b < 0: 
    
        x = x + 5 
        
    elif a > 5:
    
        x = x + 4
        
    else:
    
        x = x + 3
        
else:

    x = x + 2



### Ans: 2 is the output

In [ ]:
#...........................................................................

### 10. What will be the value of the following code :-

print( not ( not a == 10 or not b == 10) )

### Ans: True , since 'not' reverses the original result

In [54]:
#...........................................................................................................

### 11. What does the following code print out?

print("123" + "abc")
 
hello world

123+abc

This is a syntax error because you cannot add strings

123abc
### 123abc as both '123' and 'abc' are strings and '+' operator combines two strings

In [57]:
#...................................................................................................

### 12. Which of these operators is not a comparison / logical operator?
 
=,==,<,> =,!=

### Ans:'='  is not a comparison/logical operator . '='(equal to) is a Assignment operator.

In [ ]:
#.......................................................................................................

### 13. What is true about the following code segment:
if  x == 5 :
    
    print('Is 5')
    
    print('Is Still 5')
    
    print('Third 5')
 
A Depending on the value of x, either all three of the print statements will execute or none of the statements will execute

B The string 'Is 5' will always print out regardless of the value for x.

C The string 'Is 5' will never print out regardless of the value for x.

D Only two of the three print statements will print out if the value of x is less than zero.

### Ans:option A is correct i.e, Depending on the value of x, either all three of the print statements will execute or none of the statements will execute as all the statements belong to the same 'if condition statement' and executes when the condition given is true.


In [ ]:
#........................................................................................

### 14. For the following code,

if x < 2 :

    print('Below 2')
    
elif x >= 2 :

     print('Two or more')
     
else :

    print('Something else')
    
What value of 'x' will cause 'Something else' to print out?

A.	This code will never print 'Something else' regardless of the value for 'x'

B.	x = -2

C.	x = -2.0

D.	x = 2.0
### Ans: option 'A' is correct i.e "This code will never print 'Something else' regardless of the value for 'x' "( as there is no number between lesser than 2 and equal to or greater than 2)

In [ ]:
#............................................................................................

### 15,There are 100 doors in a row, all doors are initially closed. A person walks through all doors multiple times and toggles (if open then close, if close then open) them in the following way: 
In the first walk, the person toggles every door 

In the second walk, the person toggles every second door, i.e., 2nd, 4th, 6th, 8th, … 

In the third walk, the person toggles every third door, i.e. 3rd, 6th, 9th, … 

……… 
………. 


In the 100th walk, the person toggles the 100th door. 
 
What will be the position of door number 53 after this entire process ?
 
Open
Closed

### Ans: closed 
###     Reason: 53 is not a perfect square. Hence Even number of operations are operated on it.
###              Initially all doors are closed, after even number of operations the door no.53 is closed.

In [68]:
x=int(input("Enter Door number :"))
if(x**(1/2)%1==0):
   print("Door is opened")
else:
    print("Door is closed")

Enter Door number :53
Door is closed
